docs/source/index.rst

```diff
@@ -34,6 +34,9 @@ Lightning Flash
 
    general/model
    general/data
+   general/callback
```

Comment:
What do you think about creating an "Advanced" submenu here that will have all the API + creating a custom task tutorial?

Reply:
Yes, we will do so in another PR. Definitely need some re-organizations.